pkg/syncer/util.go

```diff
@@ -270,3 +275,19 @@ func initVolumeMigrationService(ctx context.Context, metadataSyncer *metadataSyn
 	}
 	return nil
 }
+
+// isPVStaticallyProvisioned checks if the PersistentVolume has been provisioned statically.
+// Dynamically provisioned PVs have a volume attribute called 'storage.kubernetes.io/csiProvisionerIdentity'
+// in their CSI spec, which is set by external-provisioner.
+// Statically provisioned PVs don't have that attribute.
+func isPVStaticallyProvisioned(pv *v1.PersistentVolume) (bool, error) {
+	if pv == nil {
+		return false, errors.New("Expected pv object not to be nil")
+	}
+	if pv.Spec.CSI != nil && pv.Spec.CSI.Driver == csitypes.Name {
```

Comment:
I'm not sure if we should have this check here. Metadata syncer and full sync already have this as a prereq in all the workflows. 
If we want to keep this check here, we need to return something else (maybe an error?) to differentiate between dynamically provisioned volumes and non vsphere CSI volumes.

Reply:
@gohilankit I think you can remove this function and move the conditional in-place, so far there is only one caller.